src/java/org/jivesoftware/openfire/plugin/rest/controller/JustMarriedController.java

```diff
@@ -237,7 +237,7 @@ private static void addNewUserToOthersRoster(User newUser, RosterItem otherItem,
 
         // Is this user registered with our OF server?
         String username = otherItem.getJid().getNode();
-        if (username != null && username.length() > 0 && userManager.isRegisteredUser(username)
+        if (username != null && username.length() > 0 && userManager.isRegisteredUser(otherItem.getJid(), true)
```

Comment:
the second argument to the invocation of `isRegisteredUser()` should be `false` to prevent lookups at remote domains. This code specifically intends to operate on local users.